src/ceph-volume/ceph_volume/devices/lvm/listing.py

```diff
@@ -140,32 +139,28 @@ def full_report(self):
         """
         return self.create_report(api.get_lvs())
 
-    def single_report(self, device):
+    def single_report(self, arg):
         """
         Generate a report for a single device. This can be either a logical
-        volume in the form of vg/lv or a device with an absolute path like
-        /dev/sda1 or /dev/sda. Returns '{}' to denote failure.
+        volume in the form of vg/lv, a device with an absolute path like
+        /dev/sda1 or /dev/sda, or a list of devices under same OSD ID.
+
+        Return value '{}' denotes failure.
         """
-        lvs = []
-        if os.path.isabs(device):
-            # we have a block device
-            lvs = api.get_device_lvs(device)
-            if not lvs:
-                # maybe this was a LV path /dev/vg_name/lv_name or /dev/mapper/
-                lvs = api.get_lvs(filters={'path': device})
+        if isinstance(arg, int) or arg.isdigit():
+            lv = api.get_lvs_from_osd_id(arg)
+        elif arg[0] == '/':
+            lv = api.get_lvs_from_path(arg)
         else:
-            # vg_name/lv_name was passed
-            vg_name, lv_name = device.split('/')
-            lvs = api.get_lvs(filters={'lv_name': lv_name,
-                                       'vg_name': vg_name})
+            lv = [api.get_first_lv(filters={'lv_name': arg.split('/')[1]})]
```

Comment:
@rishabh-d-dave would something like following work?
```suggestion
            lv = [api.get_first_lv(filters={'lv_name': arg.split('/')[1], 'vg_name': arg.split('/')[0]})]
```